engine/components-api/src/main/java/com/cloud/network/IpAddressManager.java

```diff
@@ -103,6 +103,12 @@ boolean applyIpAssociations(Network network, boolean rulesRevoked, boolean conti
 
     String acquireGuestIpAddress(Network network, String requestedIp);
 
+    String acquireFirstGuestIpAddress(Network network);
+
+    String acquireLastGuestIpAddress(Network network);
+
+    String acquireGuestIpAddressByPlacement(IpPlacement ipPlacement, Network network, String requestedIp);
```

Comment:
@ArthurHlt maybe you use create new public method without ipPlacement.

```
String acquireGuestIpAddressByPlacement(Network network, String requestedIp);
```

VrouterRedundantTiersPlacement can also be moved to IpAddressManager instead of GuestNetworkGuru